From d64fa9d8daa48347242c4a90d79ee43c67b19a38 Mon Sep 17 00:00:00 2001
From: satyam <satyamvats5@gmail.com>
Date: Wed, 13 Sep 2017 00:42:47 +0530
Subject: [PATCH] Lab_No_1: To merge two array into 3rd.

We have been given two unsorted array.
We need to merge these arrays into a third
sorted array.

Signed-off-by: satyam kumar<satyamvats5@gmail.com>
---
 2.c | 60 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 60 insertions(+)
 create mode 100644 2.c

diff --git a/2.c b/2.c
new file mode 100644
index 0000000..35303db
--- /dev/null
+++ b/2.c
@@ -0,0 +1,60 @@
+#include <stdio.h>
+int main()
+{
+	int limit1, limit2, m, n, i = 0,j = 0, k, l;
+
+	printf("Enter the array limit for 1st array\n");
+	scanf("%d",&limit1);
+	int array1[limit1];
+
+	printf("Enter the elemets of 1st array:-\n");
+	for(m = 0; m < limit1; m++) {
+		scanf("%d",&array1[m]);
+	}
+	
+	printf("ENter the array limit for 2nd array\n");
+	scanf("%d",&limit2);
+	int array2[limit2];
+	
+	printf("Enter elements of 2nd array\n");
+	for(n = 0; n < limit2;n++) {
+		scanf("%d",&array2[n]);
+	}
+		
+	int array3[limit1 + limit2];
+	
+	for( k = 0; k < limit1 +  limit2; k++) {
+		if(j<limit2 && i < limit1) {
+			if(array1[i] > array2[j]) {
+				array3[k] = array2[j];
+				j++;
+				continue;
+			} else {
+				array3[k] = array1[i];
+				i++;
+				continue;
+			}
+		}
+
+		if((j == limit2) && (i < limit1)) {
+			array3[k] = array1[i];
+			i++;
+			continue;
+		}
+		
+		if((i == limit1) && (j < limit2)) {
+			array3[k] = array2[j];
+			j++;
+			continue;
+		}
+	}
+
+	for(i = 0; i < limit1 + limit2; i++) {
+		printf("%d ",array3[i]);
+	}
+	printf("\n");
+	
+	return 0;
+} 
+
+	
-- 
2.7.4

